### PR TITLE
Make Alamofire URL Session public for easier customization

### PR DESCRIPTION
**Pull request by dejanskledar**
With this, we enable runtime dynamic customization of the default Session, after creation which was currently not possible.

### CODE REVIEW

**Review by borut-t — APPROVED** [review]
This sounds reasonable to me.

File: Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift [hunk 1]
```diff
@@ -22,7 +22,7 @@ public typealias ProgressHandler = Alamofire.Request.ProgressHandler
 public typealias ErrorHandler = (Swift.Error, Data) throws -> Swift.Error
 
 open class AlamofireNetworkClient {
-  private let session: Alamofire.Session
+  public let session: Alamofire.Session
```
Comment (tonikocjan) [inline Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift:25]: You could create a new client with a different session instead. Would that work for your use-case?

Comment (dejanlatecheckout) [reply Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift:25]: That was the initial idea, but for our use case its not possible without changing the core logic of our Network Layer / Package. :/

Comment (borut-t) [reply Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift:25]: Thinking.... 🤔
We could eventually inject required state to the project package instead, and go without this as well. What do you think @dejanlatecheckout?

Comment (dejanlatecheckout) [reply Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift:25]: This would work, let me try it out.

**Comment by dejanskledar** [comment]
Found a workaround, closing.

**Pull request closed without merging**